docs/source/user/saml_login.rst

```diff
@@ -0,0 +1,98 @@
+=====================
+Single Sign-On (SAML)
+=====================
+
+SAML login is supported by generating an additional temporary token right
+after users authenticates via SSO, the user is then redirected to the
+captive page with two querystring parameters: ``username`` and ``token``.
+
+The captive page must recognize these two parameters and automatically perform
+the submit action of the login form: ``username`` should obviously used for the
+username field, while ``token`` should be used for the password field.
+
+The internal REST API of openwisp-radius will recognize the token and authorize
+the user.
+
+This kind of implementation allows to implement the SAML login with any captive
+portal which already supports the RADIUS protocol because it's totally transparent
+for it, that is, the captive portal doesn't even know the user is signing-in with
+a SSO.
+
+.. note::
+   If you're building a public wifi service, we suggest
+   to take a look at `openwisp-wifi-login-pages <https://github.com/openwisp/openwisp-wifi-login-pages>`_,
+   which is built to work with openwisp-radius.
+
+Setup
+-----
+
+Install required dependencies::
+
+    pip install openwisp-radius[saml]
+
+Ensure your ``settings.py`` looks like the following:
+
+.. code-block:: python
+
+    INSTALLED_APPS = [
+        # ... other apps ..
+        # apps needed for SA<L> login
+        'rest_framework.authtoken',
+        'django.contrib.sites',
+        'allauth',
+        'allauth.account',
+        'djangosaml2'
+    ]
+
+    SITE_ID = 1
+
+    # Update AUTHENTICATION_BACKENDS
+    AUTHENTICATION_BACKENDS = (
+        'openwisp_users.backends.UsersAuthenticationBackend',
+        'djangosaml2.backends.Saml2Backend', # <- add for SAML login
+    )
+
+    # Update MIDDLEWARE
+    MIDDLEWARE = [
+        # ... other middlewares ...
+        'djangosaml2.middleware.SamlSessionMiddleware',
+    ]
+
+Ensure your main ``urls.py`` contains the ``allauth.urls``:
+
+.. code-block:: python
+
+    urlpatterns = [
+        # .. other urls ...
+        path('accounts/', include('allauth.urls')),
+    ]
```

Comment:
What about `(r'saml2/', include('djangosaml2.urls')),`?

Reply:
We don't need it because I wrote the all the routing in https://github.com/openwisp/openwisp-radius/blob/8fdb24b1079bf0e52cb0482d2127b51a0481d0cb/openwisp_radius/saml/urls.py since we extended views

Reply:
But without that, while testing, something wasn't working. Are you extra sure this is not needed?

Reply:
Yes! I am absolutely sure. `djangosaml2` should be present in `INSTALLED_APPS` and the URLs get added.